code/src/main/java/com/onlinecontacttracing/messaging/EmailSender.java

```diff
@@ -0,0 +1,100 @@
+package com.onlinecontacttracing.messaging;
+
+import com.onlinecontacttracing.messaging.MessageServlet;
+import com.onlinecontacttracing.messaging.SystemMessage;
+import com.onlinecontacttracing.messaging.LocalityResource;
+import com.onlinecontacttracing.messaging.MessagingSetup;
+import com.onlinecontacttracing.storage.CustomizableMessage;
+import com.onlinecontacttracing.storage.PositiveUser;
+import com.onlinecontacttracing.storage.PotentialContact;
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.extensions.java6.auth.oauth2.AuthorizationCodeInstalledApp;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.extensions.jetty.auth.oauth2.LocalServerReceiver;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.GmailScopes;
+import com.google.api.services.gmail.model.Label;
+import com.google.api.services.gmail.model.ListLabelsResponse;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.List;
+import java.util.ArrayList;
+import javax.mail.internet.MimeMessage;
+
+public class EmailSender {
```

Comment:
So, Let's break this class up into two parts:
EmailSender takes in a credential and a Positive User, and constructs a GMail Service and stores the user.

Message takes in SystemMessage systemMessage, LocalityResource localityResource, CustomizableMessage customizableMessage and MessageLanguage.

Then EmailSender has one method sendEmails(Message), and the Message class has one method getAsString().

The reason this is a better pattern is that the message properties are better captured as a part of a message object, and the sender properties are better captured as a part of the sender object.